free-1.md

```diff
@@ -11,41 +11,100 @@ _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help
 them?_
 
+The need is to improve the ease of creating crochet patterns, and to allow
+people who are currently working on a crochet project to easily keep track of
+the progress that they have made on a project.
+
 ### Why a language?
 
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+In terms of creating a crochet pattern, it would be very benificial for a
+pattern creater to have a programming language where they could easily add to,
+and manipulate the pattern, and then have an easy way to create a beautiful
+pattern output. Then, for someone using the pattern it would be easy
+to follow. There could be both a PDF output and an interative way to state
+what progress you have made in the project while following the pattern. You
+could also add your own notes, or edit the pattern yourself using the language.
+
 ### Why you?
 
 _What excites you about this idea? How did you come up with it?_
 
+I like to crochet, and I find it difficult to keep track of where I am in my
+project when I am either following a pattern, partially following a pattern
+while making my own changes, or when I have created my own projects without a
+pattern. I came up with this idea because I would like there to be an easy way
+to create crochet patterns, and I am already knowledgeable about this domain. It
+is also a cool domain because it already has it's own abbreviations (it's own
+language) that could be used in the language.
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+Crochet pattern creation and use.
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+I imagine three different ways to interact with it. First, there would be the
+coding that a user could do to create a pattern. Then, this coding could be
+exported to either a PDF, or to an interactive program where the user gets to
+keep track of what instruction they are on. I think the coding is the right
+domain because it would allow the user to easily edit and create a pattern.
+They could also add stylistic choices for the PDF output. The PDF output is
+appropriate because that is how crochet patterns are currently circulated. And
+the interactive program that you load the code into would be very helpful for
+people working on crochet projects.
+
```

Comment:
One thing you might want to consider is the overlap of the crocheting community and the CS community. That is, how familiar with coding will the typical knitter be? I suspect that most will not be very proficient, which means that a more visual GUI may be helpful (for example, you could do a drag-and-drop kind of style like Scratch). I will admit that this requires more technical work, however, which may go beyond the scope of this class.